openstack/tools/allocate_vips.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+# Allocate the last 20 IP addresses, which could be considered for
+# VIPs in OpenStack
+
+source ~/novarc
+num_vips=20
+subnet="subnet_${OS_USERNAME}-psd"
+
+cidr=$(openstack subnet show ${subnet} -c cidr -f value 2>/dev/null)
+
+net_end=$(awk -F'.' '/HostMax/{print $NF}' <<<$(ipcalc -b $cidr))
+
+net_end_suffix=${net_end##*\.}
+
+vip_start_suffix=$((net_end_suffix - ${num_vips} + 1))
```

Comment:
Parameter expansion is performed in arithmetic expansion too, so you can drop the PE: `vip_start_suffix=$((net_end_suffix - num_vips + 1))`